sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -617,3 +617,31 @@ case class DecodeUsingSerializer[T](child: Expression, tag: ClassTag[T], kryo: B
 
   override def dataType: DataType = ObjectType(tag.runtimeClass)
 }
+
+case class InitializeJavaBean(n: NewInstance, fields: Seq[Expression], setters: Seq[String])
```

Comment:
We could probably just extend [this logic](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala#L265) to recurse into maps similar to how we handle traversables.